dms-domain/src/main/kotlin/team/aliens/dms/domain/remain/service/CheckAccessibleRemainAvailableTimeService.kt

```diff
@@ -0,0 +1,25 @@
+package team.aliens.dms.domain.remain.service
+
+import team.aliens.dms.common.annotation.DomainService
+import team.aliens.dms.domain.remain.model.RemainAvailableTime
+import java.time.LocalDateTime
+
+@DomainService
+class CheckAccessibleRemainAvailableTimeService : CheckAccessibleRemainAvailableTime {
+
+    override fun execute(availableTime: RemainAvailableTime): Boolean {
+        val currentDateTime = LocalDateTime.now()
+        val dayOfWeek = currentDateTime.dayOfWeek.value
+        val now = currentDateTime.toLocalTime()
+
+        if (dayOfWeek < availableTime.startDayOfWeek.value || dayOfWeek > availableTime.endDayOfWeek.value) {
+            return false
+        }
+
+        if (now < availableTime.startTime || now > availableTime.endTime) {
+            return false
+        }
```

Comment:
```suggestion
        if (dayOfWeek == avliableTime.startDayOfWeek && now < availableTime.startTime) {
            return false
        }
        
       if (dayOfWeek == avliableTime.endDayOfWeek && now < availableTime.endTime) {
            return false
        }
```
p1 기존 코드는 잔류 신청 가능 시작요일과 종료요일이 다른 경우에 에러날 것 같습니당

Reply:
조건 하나로 통일시켜서 조회해야 할 것 같습니다.

Reply:
조건문을 하나로 줄이라는 말씀이신 건가요?

Reply:
이렇게 조회하면 목요일 10시에 신청 가능해지는거 아닌가요? `dayOfWeek == avliableTime.startDayOfWeek`이나 `dayOfWeek == avliableTime.endDayOfWeek`에 안걸리면 false로 return될 일이 없으니까요

1. 요일이 범위 밖인 경우 false
2. 시작 요일인 경우 시작 시간 이전이면 false
3. 종료 요일인 경우 종료 시간 이후면 false

이렇게 하면 딱 깔끔하게 조건문 3개로 정리될 것 같아요

Reply:
현재 로직으로는 시작 요일 전에도 가능할 것 같네요.
은빈님이 말씀해주신 3가지로 나누어서 처리하도록 변경해야할 것 같아요.